internal/session/handle_idle.go

```diff
@@ -73,3 +70,36 @@ func (s *Session) handleIdle(ctx context.Context, tag string, cmd *proto.Idle, c
 		}
 	})
 }
+
+func sendMergedResponses(s *Session, buffer []response.Response) {
+	for _, res := range response.Merge(buffer) {
+		if err := res.Send(s); err != nil {
+			logrus.WithError(err).Error("Failed to send IDLE update")
+		}
+	}
+}
+
+func sendResponsesInBulks(s *Session, resCh chan response.Response) {
+	buffer := []response.Response{}
+	ticker := time.NewTicker(500 * time.Millisecond)
+
```

Comment:
can you defer a ticker stop here so we are guaranteed to stop the ticker even if something panics?